File: content/snippets/typescript/design_patterns/multiton.ts
class Multiton {
  private static instances: Record<string, Multiton> = {}
  private constructor(public key: string){}
  static get(key: string){
    return this.instances[key] ??= new Multiton(key)
  }
}
console.log(Multiton.get('a').key, Multiton.get('b').key)
export {}

File: content/snippets/python/design_patterns/multiton.py
class Multiton:
    _instances = {}
    def __init__(self, key):
        self.key = key
    @classmethod
    def get(cls, key):
        if key not in cls._instances:
            cls._instances[key] = Multiton(key)
        return cls._instances[key]

if __name__ == '__main__':
    a = Multiton.get('a')
    b = Multiton.get('b')
    print(a.key, b.key)
